packages/better-php-doc-parser/src/PhpDocNode/Symfony/Validator/Constraints/AssertChoiceTagValueNode.php

```diff
@@ -71,4 +88,41 @@ public function getShortName(): string
     {
         return '@Assert\Choice';
     }
+
+    private function isClassConstantReference(?string $originalChoiceContent): bool
+    {
+        if ($originalChoiceContent === null) {
+            return false;
+        }
+
+        return Strings::contains($originalChoiceContent, '::');
+    }
+
+    private function createOriginalChoiceContent()
+    {
+        $content = '';
+        if ($this->hasChoiceKeywordExplicit) {
+            $content = 'choices=';
+        }
+
+        return $content . $this->originalChoiceContent;
+    }
+
+    private function resolveOriginalChoiceContent(string $annotationContent): ?string
+    {
+        $matches = Strings::match($annotationContent, '#(\(choices=|\()(?<choice_content>.*?)(\)|,)#ms');
```

Comment:
I'm not sure `choices` must be hardcoded since it is related to my use-case and it may not work for others.

Reply:
How would you implement it better?

Reply:
https://getrector.org/demo/54a3163e-13d1-4a01-8101-f32340bcd095 I changed `choices ` to `groups` and issue is back as expected. I would try to remove `choices` from regexp and detect `=` sign. If `=CONST`.

Reply:
Could you send PR?

Reply:
I'm not familiar with how Rector works under the hood. I can provide use cases and do manual testing. My point is not to hardcode `choices` because variable name could be any other or not set.

Reply:
This part is not in Rector, but in doctrine/annotations.
The best would be to disable constant value resolution in there.
This is just a bad band-aid on doctrine design choice.

If we can somehow disable this line: https://github.com/doctrine/annotations/blob/904dca4eb10715b92569fbcd79e201d5c349b6bc/lib/Doctrine/Common/Annotations/DocParser.php#L981
and `return $identifier;` instead, it would solve all these cases.

Reply:
I suggest reverting this Pull Request, reopen an issue and think about solving this issue together. May be Rector can parse a file, find annotations with constants and exclude them with `setIgnoredAnnotationNames` https://github.com/doctrine/annotations/blob/904dca4eb10715b92569fbcd79e201d5c349b6bc/lib/Doctrine/Common/Annotations/DocParser.php#L257 ?

Reply:
Here is a test https://github.com/doctrine/annotations/blob/dc296632db0500c7bd5144161cf96708f61059ff/tests/Doctrine/Tests/Common/Annotations/DocParserTest.php#L1055

Reply:
It seems unrelated.
Could you update your annotation code with that?

Reply:
Sorry, I didn't get your point. What should I do? Please provide more details.

Reply:
Working solution for your use case with "choice" annotation.

Any gist would do.

Reply:
The `setIgnoredAnnotationNames` method doesn't work. Even Doctrine's test doesn't work properly. There is a patch that looks relevant but it is not released yet. https://github.com/doctrine/annotations/pull/244

Reply:
Opened an issue to find it out https://github.com/doctrine/annotations/issues/312

Reply:
How did you verify this would help in our case?

Any reproducible code would help